components/org.wso2.carbon.identity.conditional.auth.functions.choreo/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/choreo/internal/ChoreoFunctionServiceComponent.java

```diff
@@ -0,0 +1,153 @@
+
+        BundleContext bundleContext = context.getBundleContext();
+
+        ChoreoAxis2ConfigurationContextObserver observer = new ChoreoAxis2ConfigurationContextObserver();
+        bundleContext.registerService(Axis2ConfigurationContextObserver.class.getName(), observer, null);
+        ServerConfigurationService config = ChoreoFunctionServiceHolder.getInstance()
+                .getServerConfigurationService();
+
+        String filePath = config.getFirstProperty("Security.TrustStore.Location");
+        String keyStoreType = config.getFirstProperty("Security.TrustStore.Type");
+        String password = config.getFirstProperty("Security.TrustStore.Password");
+        try (InputStream keyStoreStream = new FileInputStream(filePath)) {
+            KeyStore keyStore = KeyStore.getInstance(keyStoreType); // or "PKCS12"
+            keyStore.load(keyStoreStream, password.toCharArray());
+            ChoreoFunctionServiceHolder.getInstance().setTrustStore(keyStore);
+        } catch (IOException | CertificateException | KeyStoreException | NoSuchAlgorithmException e) {
+            LOG.error("Error while loading truststore.", e);
+        }
+    }
+
+    @Deactivate
+    protected void deactivate(ComponentContext ctxt) {
+
+        JsFunctionRegistry jsFunctionRegistry = ChoreoFunctionServiceHolder.getInstance().getJsFunctionRegistry();
+        if (jsFunctionRegistry != null) {
+            jsFunctionRegistry.deRegister(JsFunctionRegistry.Subsystem.SEQUENCE_HANDLER, FUNC_CALL_CHOREO);
+        }
+    }
+
+    @Reference(
+            service = JsFunctionRegistry.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetJsFunctionRegistry"
+    )
+    public void setJsFunctionRegistry(JsFunctionRegistry jsFunctionRegistry) {
+
+        ChoreoFunctionServiceHolder.getInstance().setJsFunctionRegistry(jsFunctionRegistry);
+    }
+
+    public void unsetJsFunctionRegistry(JsFunctionRegistry jsFunctionRegistry) {
+
+        ChoreoFunctionServiceHolder.getInstance().setJsFunctionRegistry(null);
+    }
+
+    @Reference(
+            name = "identityCoreInitializedEventService",
+            service = IdentityCoreInitializedEvent.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetIdentityCoreInitializedEventService")
+    protected void setIdentityCoreInitializedEventService(IdentityCoreInitializedEvent identityCoreInitializedEvent) {
+
+    /* reference IdentityCoreInitializedEvent service to guarantee that this component will wait until identity core
+         is started */
+    }
+
+    protected void unsetIdentityCoreInitializedEventService(IdentityCoreInitializedEvent identityCoreInitializedEvent) {
+
+    /* reference IdentityCoreInitializedEvent service to guarantee that this component will wait until identity core
```

Comment:
what is the intension of this comment?

Reply:
Shall we make this a method level comment or properly format it as a single line comment?

Reply:
Formatted with commit [97b8dc8](https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/85/commits/97b8dc822153f28e50e8f878d46d78b6d6dfd1c0)